src/main/java/StationsGraph.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class StationsGraph {
+    private final HashMap<String, StationNode> stationNodes;
+
+    /**
+     * Constructor creates all stationNodes in the graph
+     * @param stations Hashmap
+     */
+    public StationsGraph(HashMap<String, List<String>> stations) {
+        this.stationNodes = new HashMap<>();
+        //for each id of the stations hashmap, it creates a stationNode with that id
+        //and then puts the id and stationNode in the stationNodes hashMap
+        for(String stationId: stations.keySet()){
+            stationNodes.put(stationId, new StationNode(stationId));
+        }
+        //connecting stationNodes to each other
+        //for each id in stations, it stores the neighbors of that stationId in a List
+        for(String stationId : stations.keySet()){
+            List<String> NeighborIds = stations.get(stationId);
+            //connects the neighboring nodes to walk over the graph
+            //for each neighborId in the NeighborIds list, adds stationNode of that neighborId to the stationNodes list
+            for(String neighborId : NeighborIds){
+                stationNodes.get(stationId).neighbors.add(stationNodes.get(neighborId));
```

Comment:
I think you're adding all the stations on each line as neighbors.
Neighbors are **ONLY** direct connections of a Station